### ISSUE TITLE
Getting "docker: Error response from daemon: Mounts denied: EOF." in MacOs Catalina

### ISSUE THREAD

**Issue by mottolini** [nginx-proxy/nginx-proxy#1458]
I'm using Docker Desktop 2.3.0.2 on MacOs Catalina 10.15.4 and I'm trying to run the nginx-proxy with:

    docker run -d --name nginx -p 8087:80 -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy

But I receive the error `docker: Error response from daemon: Mounts denied: EOF.`
I set /var/run in the Preferences/Resources/FileSharing and tried to run it with sudo, but error temains the same.
Is there any specific setting I'm missing?

**Comment by tkw1536** [comment]
This sounds like a docker for mac issue. Docker for Desktop on mac does not run docker directly. Instead, it starts up a linux virtual machine in the background to run the docker daemon in. To enable mounting real files and folders inside containers, they are first mounted into this VM and then into the containers. The paths that can be mounted can be configured via the `FileSharing` option. 

Unlike normal file mounts, the docker the docker socket we want to mount only lives in that VM.  Through various tricks it is also mounted on the real mac system. Adding '/var/run' to `Preferences/Resources/FileSharing` probably confuses the docker daemon -- because both the VM and real system are trying to write to it. Try removing `/var/run` from the FileSharing options. You might have to restart docker for any changes to take effect. 

I've just tested running your command line on my mac with the default shared folder settings, and it works fine. However when I added `/var/run` to the shared files and was able to reproduce your issue.
